libyul/YulString.h

```diff
@@ -71,10 +71,10 @@ class YulStringRepository
 	{
 		// FNV hash - can be replaced by a better one, e.g. xxhash64
 		std::uint64_t hash = emptyHash();
-		for (auto c: v)
+		for (char c: v)
 		{
 			hash *= 1099511628211u;
-			hash ^= c;
+			hash ^= uint64_t(c);
```

Comment:
Does the conversion here work for both `char` being signed and unsigned?

Reply:
I changed to named cast, it should work fine for both